### DISCUSSION TITLE
Coursier API - almond bootstrap?

### DISCUSSION THREAD

**Discussion by Quafadas** [coursier/coursier#2479] [Q&A] [answered]
I've gotten myself in over my head trying to look at this; 
https://github.com/scalameta/metals-feature-requests/issues/236

I've been reading the docs for bootstrapping / installing almond, all of which go through the coursier CLI. If I wished to bootstrap and install and almond instead through the coursier API, for example in the context of metals, is that possible? And if so, a hint at the starting point?

**Comment by alexarchambault** [comment, marked as answer]
If you wish to create an Almond launcher from the API, the [ammonite-runner](https://github.com/alexarchambault/ammonite-runner) code base should be a good example. [This section](https://github.com/alexarchambault/ammonite-runner/blob/5d2ec0f6c7d430b59f8f66d578099b711a6bf34c/core/src/main/scala/ammrunner/AmmoniteFetcher.scala#L143-L170) in particular. It shows how to create a launcher with some shared dependency put in a first class loader by the launcher, and remaining dependencies put in a second class loader, as the installation instructions of Almond do via [this app descriptor](https://github.com/coursier/apps/blob/1a067dc6d46c3dfbc013b1b7bdc6adb06062fbd7/apps/resources/almond.json).
